src/main/java/com/coveros/selenified/utilities/Property.java

```diff
@@ -153,13 +162,39 @@ public static boolean isProxySet() {
      *
      * @return String: the set proxy address, null if none are set
      */
-    public static String getProxy() {
+    public static String getProxy() throws InvalidProxyException {
         if (!isProxySet()) {
-            return null;
+            throw new InvalidProxyException(PROXY_ISNT_SET);
+        }
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "proxy" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS2259)